### PR TITLE
V3 Cosmos Code Migration

### PR DESCRIPTION
**Pull request by bryevdv**
This migration is a straight copy/add and does not preserve v3 history.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by bryevdv** [comment]
Once this is approved/merged I will make a new PR to update the master branch to this same direct download/install process.

File: sdk/cosmos/.azure-pipelines/cosmos.test.yml [hunk 1]
```diff
@@ -26,27 +26,67 @@ jobs:
...
+      - task: CmdLine@2
+        displayName: 'Install Public Cosmos DB Emulator'
+        inputs:
+          script: |
+            choco install lessmsi
+            choco upgrade lessmsi
+            echo "Checking directory" 
+            dir "%ProgramFiles%"
+            mkdir "%TEMP%\Azure Cosmos DB Emulator"
+            lessmsi x "%TEMP%\azure-cosmosdb-emulator.msi" "%TEMP%\Azure Cosmos DB Emulator\"
```
Comment (weshaggard) [inline sdk/cosmos/.azure-pipelines/cosmos.test.yml]: It seems like you are simply extracting files from the msi, is there a reason to not use msiexec instead?

File: sdk/cosmos/.azure-pipelines/cosmos.test.yml [hunk 2]
```diff
@@ -26,27 +26,67 @@ jobs:
       vmImage: $(OSVmImage)
 
     steps:
+      - task: PowerShell@1
+        displayName: 'Download Public Cosmos Emulator'
+        inputs:
+          scriptType: inlineScript
+          inlineScript: |
+            Write-Host "Downloading Cosmos Emulator - $(EmulatorMsiUrl)"
+            wget "$(EmulatorMsiUrl)" -outfile "$env:temp\azure-cosmosdb-emulator.msi"
```
Comment (weshaggard) [inline sdk/cosmos/.azure-pipelines/cosmos.test.yml]: May I ask why we are now consuming this via an msi?

File: sdk/cosmos/.azure-pipelines/cosmos.test.yml [hunk 3]
```diff
@@ -26,27 +26,67 @@ jobs:
...
+            lessmsi x "%TEMP%\azure-cosmosdb-emulator.msi" "%TEMP%\Azure Cosmos DB Emulator\"
+            dir "%TEMP%"
+            dir "%Temp%\Azure Cosmos DB Emulator\"
+
+      - task: PowerShell@1
+        displayName: 'Run Public Cosmos DB Emulator'
+        inputs:
+          scriptType: inlineScript
+          inlineScript: |
+            dir "$env:Temp\"
```
Comment (weshaggard) [inline sdk/cosmos/.azure-pipelines/cosmos.test.yml]: Why are we doing a directory listing of all these directories?

File: sdk/cosmos/.azure-pipelines/cosmos.test.yml (hunk 2 above)
Comment (bryevdv) [reply sdk/cosmos/.azure-pipelines/cosmos.test.yml]: Evidently there are problems using the emulator with v3 currently, and it cannot be used. This process is copied from https://cosmos-db-sdk-public.visualstudio.com/cosmos-db-sdk-public/_build?definitionId=32

File: sdk/cosmos/.azure-pipelines/cosmos.test.yml (hunk 1 above)
Comment (bryevdv) [reply sdk/cosmos/.azure-pipelines/cosmos.test.yml]: I don't have any experience to bear on this question, currently have only duplicated what is at https://cosmos-db-sdk-public.visualstudio.com/cosmos-db-sdk-public/_build?definitionId=32

File: sdk/cosmos/.azure-pipelines/cosmos.test.yml (hunk 3 above)
Comment (bryevdv) [reply sdk/cosmos/.azure-pipelines/cosmos.test.yml]: Same answer as above, this came from the original build on the cosmos side. Happy to remove everyone agrees.

**Comment by bryevdv** [comment]
@weshaggard FYI I will need to do some track 2 work off of the the cosmos_v3 branch ASAP when I get back next week. What do you think about merging this now and making some follow up issues for later?

**Comment by weshaggard** [comment]
>@weshaggard FYI I will need to do some track 2 work off of the the cosmos_v3 branch ASAP when I get back next week. What do you think about merging this now and making some follow up issues for later?

Sounds good.

**Pull request merged**